README.md

```diff
@@ -129,3 +61,232 @@ const stream = Kafka.KafkaConsumer.createReadStream(
   }
 );
 ```
+
+The parameters for `getBinding` include:
+
+| Parameter                         | Type     |
+| --------------------------------- | -------- |
+| [type](#type)                     | `String` |
+| [client](#client)                 | `String` |
+| [bindingOptions](#bindingoptions) | `Object` |
+
+### type
+
+The type of service for which a binding is being
+requested. Currently the supported types are:
+
+- 'KAFKA'
+- 'POSTGRESQL'
+- 'REDIS'
+- 'MONGODB'
+- 'AMQP'
+- 'MYSQL'
+
+### client
+
+The package the application is using to connect
+to the service. kube-service-bindings is aware of a
+subset of possible packages. For those that it is aware
+of, it can map the service bindings into the form
+required by the client.
+
+Currently the following clients are recognized based on the supported types:
+
+- KAFKA
+  - node-rdkafka
+  - kafkajs
+- POSTGRESQL
+  - pg
+  - odbc
+- REDIS
+  - redis
+  - ioredis
+- MONGODB
+  - mongodb
+  - mongoose
+- AMQP
+  - rhea
+- MYSQL
+  - mysql
+  - mysql2
+  - odbc
+
+**(Deprecated)** If you don't specify a client, the object returned will
+be a direct map from the bindings, with the keys
+corresponding to the name of each file provided by the
+binding.
+
+#### Example on mongoDB client
+
+```javascript
+const serviceBindings = require('kube-service-bindings');
+const { MongoClient } = require('mongodb');
+
+const { url, connectionOptions } = serviceBindings.getBinding(
+  'MONGODB',
+  'mongodb'
+);
+
+const mongoClient = new MongoClient(url, connectionOptions);
+```
+
+### bindingOptions
+
+An object where provides control on parsing binding data.
+
+| Attribute                         | type      | default Value |
+| --------------------------------- | --------- | ------------- |
+| [id](#id)                         | `String`  | undefined     |
+| [removeUnmapped](#removeunmapped) | `Boolean` | true          |
+| [allwCopy](#allowcopy)            | `Boolean` | false         |
+| [bindingData](#bindingdata)       | `Object`  | undefined     |
+
+#### id
+
+Option id that is used to filter the available bindings in the search. Most useful for testing where you might have more than one binding of a given type available.
+
+#### removeUnmapped
+
+Fine grains parsed binding data to fit for the specified client, removing unnecessary properties from the input object.
```

Comment:
```suggestion
Removes all binding data which is not mapped for the client. Otherwise any binding data which is not mapped remains in it's raw for on the binding object returned.  The default depends on the client.
```